tests/IntegrationTest/BaseLineTest.php

```diff
@@ -0,0 +1,167 @@
+ * @license MIT
+ */
+
+namespace Dealerdirect\Composer\Plugin\Installers\PHPCodeSniffer\Tests\IntegrationTest;
+
+use Dealerdirect\Composer\Plugin\Installers\PHPCodeSniffer\Tests\PHPCSVersions;
+use Dealerdirect\Composer\Plugin\Installers\PHPCodeSniffer\Tests\TestCase;
+
+/**
+ * Test baseline.
+ *
+ * These tests verify:
+ * - That the plugin can be installed and functions correctly with the full range of supported PHPCS versions.
+ *   While the test is not run against the full range of supported PHPCS version each time, due to the
+ *   random PHPCS version selection, all supported PHPCS versions will be tested over time in CI.
+ * - That the plugin runs when installed.
+ * - That the PHPCS native standards are the only recognized standards when no external standards are available.
+ * - That no `CodeSniffer.conf``file gets created when no external standards are found.
+ */
+final class BaseLineTest extends TestCase
+{
+    private $composerConfig = array(
+        'name'        => 'phpcs-composer-installer/baseline-test',
+        'require-dev' => array(
+            'squizlabs/php_codesniffer'                      => null,
+            'dealerdirect/phpcodesniffer-composer-installer' => '*',
+        ),
+    );
+
+    /**
+     * Set up test environment before each test.
+     */
+    protected function set_up()
+    {
+        $this->createTestEnvironment();
+    }
+
+    /**
+     * Clean up after each test.
+     */
+    protected function tear_down()
+    {
+        $this->removeTestEnvironment();
+    }
+
+    /**
+     * Baseline test for a Composer GLOBAL install.
+     *
+     * @dataProvider dataBaseLine
+     *
+     * @param string $phpcsVersion  PHPCS version to use in this test.
+     *                              This version is randomly selected from the PHPCS versions compatible
+     *                              with the PHP version used in the test.
+     * @param array  $expectedStnds List of the standards which are expected to be registered.
+     *
+     * @return void
+     */
+    public function testBaseLineGlobal($phpcsVersion, $expectedStnds)
+    {
+        $config                                             = $this->composerConfig;
+        $config['require-dev']['squizlabs/php_codesniffer'] = $phpcsVersion;
+
+        $this->writeComposerJsonFile($config, static::$tempGlobalPath);
+        $this->assertComposerValidates(static::$tempGlobalPath);
+
+        // Make sure the plugin runs.
+        $expectedStdOut = $this->willPluginOutputShow() ? 'Running PHPCodeSniffer Composer Installer' : null;
+        $this->assertExecute(
+            'composer global install -v --no-ansi',
+            0,    // Expected exit code.
+            $expectedStdOut, // Expected stdout.
+            null, // No stderr expectation.
+            'Failed to install dependencies.'
+        );
+
+        $result = $this->executeCliCommand('"vendor/bin/phpcs" -i', static::$tempGlobalPath);
```

Comment:
If the method is called from within an instantiated class which extends `TestCase`, it doesn't matter, all three (four if we also count `parent::`) ways of calling the method are perfectly valid.

It only matters when the method is used outside of an instantiated class. In that case, only `self::`, `parent::` or `static::` are available as `static` methods cannot be called non-statically outside of an instantiated class.

The `executeCliCommand()` method was made static to allow it to be used in `setUpBeforeClass()` methods too and when used in those, it can only be called statically.

On that note - the same can be said about the PHPUnit native/polyfilled assertion calls as those are all defined as static methods too, but commonly called using `$this->assert...()` (including in this test suite).